### PR TITLE
Simplify Firehose logging interface

### PR DESCRIPTION
**Pull request by ewjordan**
Removes the stream parameter from Firehose logging calls, since the only stream we support is `analysis-events` (at least without AWS-level changes).

I have not tested this yet, so it's a WIP. Reviews requested for feedback, will likely be making further changes.

I'm not opposed to supporting additional streams in the client constructors, but at the moment it's a pain to create these manually, so I'd rather default to the one that we actually support.

### CODE REVIEW

**Comment by davidsbailey** [comment]
Hi Eric, thanks for addressing this. It seems like the main considerations here are (1) preventing careless errors when adding putRecord calls (I just messed this up yesterday in a way this PR would have prevented), and (2) eliminating overly general code.

What you have here seems like the best approach, unless we are fairly sure we are going to be supporting more stream types soon. If we are going to be adding new streams soon, then one alternative would be to keep the parameters as they are and add some validation logic, so that when we make this call in the development environment, we get some immediate feedback about whether we've formatted our putRecord call correctly. Then when we go to add more streams types, we can just expand the validation logic to allow the new types. 

One benefit of adding validation logic would be that it could catch other types of errors, such as omitting required fields or specifying other fields which are not valid optional fields.

**Comment by asherkach** [comment]
FWIW, when I created the API, I assumed we would be adding more streams (with accompanying tables in Redshift). Since that hasn't happened, I totally agree the parameter should be eliminated.

File: lib/cdo/firehose.rb [hunk 1]
```diff
@@ -18,6 +17,9 @@
 #       data_json: "{\"key\":\"value\"}"    # OPTIONAL
 #     }
 #   )
+
+STREAM_NAME = 'analysis-events'
```
Comment (asherkach) [inline lib/cdo/firehose.rb]: nit: Use `freeze` on constants.

Comment (davidsbailey) [reply lib/cdo/firehose.rb]: Thanks for continuing to help us improve our code hygiene, @ashercodeorg !  I also saw a big PR from you adding `.freeze` to a bunch of places. What is the long term vision here? Ruby 2.3 supports the `frozen_string_literal` comment and command line flag, and also Ruby 3.0 will freeze string literals by default.

Comment (asherkach) [reply lib/cdo/firehose.rb]: My long term vision is to reenable the rubocop `Style/MutableConstant` cop. Though freezing strings may have limited value, freezing other data types has definite value (has caught bugs).

**Pull request merged**